activiti-cloud-starter-runtime-bundle/src/test/java/org/activiti/cloud/starter/tests/runtime/ProcessVariablesIT.java

```diff
@@ -326,4 +336,360 @@ public void adminShouldSeeVariables() {
                                              variableCollection)).isTrue();
         });
     }
+    
+    @Test
+    public void shouldProperHandleProcessVariablesForAdmin() throws Exception {
+        //given
+        keycloakSecurityContextClientRequestInterceptor.setKeycloakTestUser("hradmin");
+        checkProcessVariables(true);
+    }
+    
+    @Test
+    public void shouldProperHandleProcessVariables()  throws Exception {
+        //given
+        keycloakSecurityContextClientRequestInterceptor.setKeycloakTestUser("hruser");
+        checkProcessVariables(false);
+    }
+    
+    public void updateSimpleVariables(boolean isAdmin,
+                                      String processInstanceId) {
+        Map<String, Object> variables = new HashMap<>();
+        
+        variables.put("variableInt",
+                      2);
+        variables.put("variableStr",
+                      "new value");
+        variables.put("variableBool",
+                      false);
+ 
+        if (isAdmin) {
```

Comment:
This `if`/`else` can be avoided if you use two separeted methods, one for admin and one for user.

Reply:
I have tried to reduce use of if, but also I do not want to have a duplication of code... I have made some refactoring, but still I think that use of if statement is better in this test than use of another separate method